### ISSUE TITLE
Plack::Runner watched for non-existing directories (watch/autorestart does not work)

### ISSUE THREAD

**Issue by tomas-zemres** [plack/Plack#622]
If I run my webapp in terminal:
   `starman  $args -R /myapp/perllib   /myapp/bin/app.psgi`

It says:
```
!! Using restarter with -r or -R options is known to cause issues with process supervisor
!! such as start_server, and it is STRONGLY discouraged to use with Starman. You have been warned.
!!
Watching /myapp/perllib /myapp/bin/lib /myapp/bin/app.psgi for file updates.
watch failed: No such file or directory at /opt/perl-5.26.1/lib/site_perl/5.26.1/Plack/Loader/Restarter.pm line 79.
```

There is reported error "No such file or directory"
 (there is missing filename, due to another bug - https://github.com/miyagawa/Filesys-Notify-Simple/issues/26), but missing filename is **"/myapp/bin/lib"**

But directory **"/myapp/bin/lib"** does not exist in my project.

Plack::Runner calls watch(...) on non-existing directory:
https://github.com/plack/Plack/blob/master/lib/Plack/Runner.pm#L178
    $self->loader->watch( File::Basename::dirname($psgi) . "/lib", $psgi );

"Auto-restart" is not working in this case also for other watched directories after argument "-R"

**Comment by belgh** [comment]
I have a same problem.（ここ日本語で書いていいのかな…google翻訳英語ですみません）

Startup command
`plackup /var/www/root_dir/script/index.psgi -s Starlet -R /var/www/root_dir/conf/,/var/www/root_dir/lib/`

Error occurred (The child process is repeatedly generated until the server is stopped)
`Watching /var/www/root_dir/conf/ /var/www/root_dir/lib/ /var/www/root_dir/script/lib /var/www/root_dir/script/index.psgi for file updates.
watch failed: そのようなファイルやディレクトリはありません at /usr/local/share/perl5/Plack/Loader/Restarter.pm line 79.`

In my environment, there is no lib directory under directory(/var/www/root_dir/script) of *.psgi.

In my environment there is no problem with $ Plack :: VERSION = 1.0044,
After that my environment has been rebuilt(including OS) to 1.0047, an error has come to be issued.
The contents of Runner.pm look the same.
（1.0044では問題なく動いていましたが、OSごと再構築したついでに1.0047にバージョンアップしてからエラーが発生するようになりました。Ruuner.pmの問題の箇所は同じに見えます）

**Comment by belgh** [comment]
plackup: Check the default path of watch exists #620 

A pull request was present before this report. I found just now.
Perhaps I think this issue will be fixed.

**Comment by miyagawa** [comment]
i made a temporary fix by reverting the breaking change (#21) in Filesys::Notify::Simple in 0.14
